pallets/attestation/src/lib.rs

```diff
@@ -130,15 +128,15 @@ decl_module! {
 			let sender = ensure_signed(origin)?;
 
 			// lookup attestation & check if the attestation exists
-			let mut existing_attestation = <error::Module<T>>::ok_or_deposit_err(
+			let Attestation {ctype_hash, owner, delegation_id, revoked, ..} = <error::Module<T>>::ok_or_deposit_err(
 				<Attestations<T>>::get(claim_hash),
 				Self::ERROR_ATTESTATION_NOT_FOUND
 			)?;
 			// if the sender of the revocation transaction is not the attester, check delegation tree
-			if !existing_attestation.1.eq(&sender) {
-				match existing_attestation.2 {
-					Some(d) => {
-						if !<delegation::Module<T>>::is_delegating(&sender, &d, max_depth)? {
+			if !owner.eq(&sender) {
+				match delegation_id {
+					Some(delegation_id) => {
+						if !<delegation::Module<T>>::is_delegating(&sender, &delegation_id, max_depth)? {
```

Comment:
maybe we could use an `if let some` here to reduce the indentation? Another option would be to do

```rust
let del_id = delegation_id.ok_or_else(ERROR)?;

if !is_delegating(...) {
    ERROR
}
```

I'm a fan of the later because the errors occur in code in the order they are checked

Reply:
This will be part of the error removal ticket #940!

Reply:
Ok than let's meeeeeerge! 😁